### PR TITLE
Option to set a default address validation carrier #7135

### PR DESCRIPTION
**Pull request by aroraumang**

### CODE REVIEW

File: carrier.py [hunk 1]
```diff
@@ -16,17 +18,35 @@ class CarrierConfig(ModelSingleton, ModelSQL, ModelView):
...
+        table = TableHandler(cursor, cls, module_name)
+
+        if table.column_exist('default_validation_provider'):
+            table.drop_column('default_validation_provider')
+
+    @classmethod
+    def __setup__(cls):
+        super(CarrierConfig, cls).__setup__()
+
+        carrier_domain = cls.carrier_domain()
```
Comment (tarunbhardwaj) [inline carrier.py]: not sure If this will respect inheritance

File: carrier.py [hunk 2]
```diff
@@ -16,17 +18,35 @@ class CarrierConfig(ModelSingleton, ModelSQL, ModelView):
...
+        cls.default_validation_carrier.domain = carrier_domain
```
Comment (tarunbhardwaj) [inline carrier.py]:
```
cls.default_validation_carrier.domain =  [('carrier_cost_method', 'in', carrier_domain)]
```

File: carrier.py [hunk 3]
```diff
@@ -16,17 +18,35 @@ class CarrierConfig(ModelSingleton, ModelSQL, ModelView):
...
+    @classmethod
+    def carrier_domain(cls):
         """
-        Downstream modules can implement `_<provider>_address_validation`
-        method in address and append <provider> to this list.
+        Return the domain for default validation carrier
         """
-        return [(None, '')]
+        # XXX: Create a domain
+        return []
```
Comment (tarunbhardwaj) [inline carrier.py]: just return list of `carrier_cost_method` not the complete domain

File: carrier.py [hunk 4]
```diff
@@ -16,17 +18,37 @@ class CarrierConfig(ModelSingleton, ModelSQL, ModelView):
     "Carrier Configuration"
     __name__ = 'carrier.configuration'
 
-    default_validation_provider = fields.Selection(
-        'get_default_validation_providers', 'Default Validation Provider'
+    default_validation_carrier = fields.Many2One(
+        'carrier', 'Default Validation Carrier'
```
Comment (prakashpp) [inline carrier.py]: I think this need a domain for now, restricted to the carriers which can do address validation?

Comment (prakashpp) [inline carrier.py]: sorry I didnt see the next part

**Comment by prakashpp** [comment]
LGTM

**Pull request merged**